patches/minecraft/net/minecraft/entity/Entity.java.patch

```diff
@@ -356,6 +356,57 @@
 +            && (this instanceof EntityPlayer || world.func_82736_K().func_82766_b("mobGriefing"))
 +            && this.field_70130_N * this.field_70130_N * this.field_70131_O > 0.512F;
 +    }
++
++    /**
++     * A functional call to consume a shared list of item drops, usually
++     * to collect all drops, modify, and or clear, before throwing an event.
++     * This is to be used instead of manually flipping fields. The provided
++     * list is always cleared on each call. If an event is to be thrown, the
++     * event should be thrown in the provided {@link java.util.function.Consumer}
```

Comment:
I the method can be slightly simplified to this. 
Since nothing else can access the `capturedDrops` field, I think we can be confident that it doesn't need to be cleared before each call. This method should also probably be `final` since overriding it seems like a bad idea.
The javadoc can also be simplified.

```java
    /**
     * Captures item drops from this entity.
     * Any items that are dropped inside the body of the Consumer will be added to 
     * the Consumer's List instead of dropped into the world.
     * @param droppedItems
     */
    public final void captureDrops(java.util.function.Consumer<List<EntityItem>> droppedItems)
    {
        if (this.performingDropCaptures)
        {
            droppedItems.accept(this.capturedDrops);
            return;
        }
        this.performingDropCaptures = true;
        droppedItems.accept(this.capturedDrops);
        this.capturedDrops.clear();
        this.performingDropCaptures = false;
    }
```